07-Interpretability-of-ML/LRP-L96.ipynb

```diff
@@ -1,5 +1,16 @@
 {
```

Comment:
<p>Since this is the first time I am seeing this, I am not 100% sure what equations are being referred to here. Do you mean the L-96 equations? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/m2lines/L96_demo/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
Hmm, on further reflection, my comment doesn't make sense. What I was remembering, though, was the fact that when I trained a [linear model to predict subgrid forcing](https://github.com/m2lines/team-meetings/blob/main/22-01-12-deep_vs_other_ML/symbolic_vs_nn_multiscale_L96.ipynb) (and [when Anastasia did](https://github.com/m2lines/team-meetings/blob/main/22-01-12-deep_vs_other_ML/symbolic_vs_nn_multiscale_L96.ipynb)), it learned forcing prediction coefficients for all `X_i` which were equal and about -0.8 (or alternatively, changing the corresponding differential equation weight from -1 to -1.8 in Anastasia's SINDY case).

The fact that a neural network's input gradients are all about -0.8 makes a lot of sense given those results, but I can't relate that back to the original equations, since we don't have any actual equations for the subgrid forcing in terms of the large-scale variables. Need to think about how to describe that succinctly in the notebook (+ with a reference to something public, since we can't link to m2lines presentations), let me know if you have suggestions.

Reply:
In lieu of a satisfactory physical explanation, I updated the notebook to actually train and visualize the linear model for comparison!